modules/custom-status/custom-status.php

```diff
@@ -77,8 +83,6 @@ public function init() {
 		add_action( 'admin_menu', [ $this, 'add_admin_menu' ], 6 /* Prior to default registration of sub-pages */ );
 
 		// These seven-ish methods are hacks for fixing bugs in WordPress core
-		add_action( 'admin_init', [ $this, 'check_timestamp_on_publish' ] );
```

Comment:
I ended up using the chance to clear our this core hack that was in place. Essentially, by adding `date_floaing` to our custom statuses it's able to get pulled into any of the `get_post_stati` calls as they check if the `date_floaing` is true or not. This allows the timestamp to get set correctly. To test this, you have to ensure your timezone offset is set correctly. Before I added the `date_floating`, if you took out the hacks you'd see the GMT time only be set even if your timezone was set correctly. Also, the exact time of publishing was off a bit.

With this, we should be down 2 core hacks which is nice.

Reply:
https://core.trac.wordpress.org/ticket/18362 - core ticket

https://wordpress.org/support/topic/plugin-edit-flow-custom-statuses-create-timestamp-problem/ - the bug

Reply:
Awesome!